lib/srv/sess.go

```diff
@@ -278,19 +278,17 @@ func (s *SessionRegistry) TryCreateHostUser(identityContext IdentityContext) (cr
 		return false, nil, nil // not an error to not be able to create host users
 	}
 
-	ui, err := identityContext.AccessChecker.HostUsers(s.Srv.GetInfo())
-	if err != nil {
-		if trace.IsAccessDenied(err) {
-			log.Warnf("Unable to create host users: %v", err)
-			return false, nil, nil
+	ui, accessErr := identityContext.AccessChecker.HostUsers(s.Srv.GetInfo())
+	existsErr := s.users.UserExists(identityContext.Login)
+	if existsErr != nil {
+		if trace.IsAccessDenied(accessErr) {
+			return false, nil, trace.WrapWithMessage(accessErr, "Insufficent permission for host user creation")
```

Comment:
We need to be really careful not to introduce a regression here. Previously, we returned no error if `trace.IsAccessDenied(accessErr) && existsErr != nil`, now we do. Which I know is the entire point of this PR but I think there are cases where the SSH connection should actually succeed:
- the `UserExists` implementation might return an error even though the user does, in fact, exist (it would on non-linux, but `GetCreateHostUser()` above would protect us there. Maybe there are other cases)
- the user doesn't exist now, but will be created just-in-time by a PAM module https://goteleport.com/docs/enroll-resources/server-access/guides/ssh-pam/#create-local-unix-users-on-login

I think we should probably only return the access denied error if the SSH connection actually fails because the user doesn't exist. Hopefully there's a way to catch that error and add the extra context